src/VisualStudio/Core/Def/Implementation/Extensions/DocumentExtensions.cs

```diff
@@ -59,5 +60,15 @@ internal static class DocumentExtensions
 
             return codeBlocks;
         }
+
+        public async static Task<TextSpan> GetSpanFromPositionAsync(this Document document, int startLine, int startColumn, int endLine, int endColumn, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
this doesn't work. there is no guarantee data you got from out of proc is actually valid with current solution. currently there is no way to track version of snapshot out of proc build actually used.